locallibrary/catalog/views.py

```diff
@@ -1,3 +1,26 @@
 from django.shortcuts import render
 
 # Create your views here.
+from catalog.models import Book, Author, BookInstance, Genre
+def index(request):
+    '''View function for home page of site. '''
+
+    # Generate counts of some of the main objects
+    num_books = Book.objects.all().count()
+    num_instances = BookInstance.objects.all().count()
+
+    # Available books (status = 'a' )
+    num_instances_available = BookInstance.objects.filter(status__exact = 'a').count()
```

Comment:
E tìm hiểu về các Fields lookup khác ngoài `exact`

Reply:
Dạ, 1 số Fields Lookup:
+ contains/icontains: trả về bản ghi có chứa từ khóa / không phân biệt chữ hoa và chữ thường
+ second/minute/hour/month/quarter/time/week: Match với một giây/phút/giờ/tháng/một phần tư của năm/time/tuần
+ startswith/endswith: trả về bản ghi bắt đầu/kết thúc bằng từ khóa nào đó
+ lt/lte: less than/less than or equal to
+ regex: Match với một biểu thức 
+ in: match với 1 phần tử nằm trong một tập hợp (ví dụ: list, tuple, set,..)
+ range: match với phần tử có giá trị nằm trong 1 khoảng chỉ định